include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -249,6 +292,78 @@ class fm_index
         // index.m_sigma = largest_char;
     }
 
+    //!\overload
+    void construct(text_t const & text)
+        //!\cond
+        requires is_collection
+        //!\endcond
+    {
+        // text collection must not be empty
+        if (std::ranges::begin(text) == std::ranges::end(text))
+            throw std::invalid_argument("The text that is indexed cannot be empty.");
+
+        // A text collection that contains the char with rank 255 or 254 cannot be indexed.
+        // 255 will overflow to 0 (reserved by sdsl) and 254 will be incremented to 255 (reserved as delimiter)
+        assert(!std::ranges::any_of(text | ranges::view::join,
+                                    [] (char_type r)
+                                    {
+                                        return to_rank(r) == static_cast<underlying_rank_t<char_type>>(255) ||
+                                               to_rank(r) == static_cast<underlying_rank_t<char_type>>(254);
+                                    }));
+
+        size_t concatenated_size{0};
+        // there must be at least one non-empty text in the collection
+        bool all_empty = true;
+        for (auto && t : text)
+        {
+            if (std::begin(t) != std::end(t))
+            {
+                all_empty = false;
+                concatenated_size += 1 + t.size(); // text size and delimiter
+            }
+        }
+        if (all_empty)
+            throw std::invalid_argument("A text collection that only contains empty texts cannot be indexed.");
+        this->text = &text;
+
+         // size + 1: for the rank query we need to go one to the right: rank=#1 BEFORE (i.e. excluding) the position
+        sdsl::bit_vector pos(concatenated_size, 0);
+        size_t prefix_sum{0};
+        for (auto && t : text)
+        {
+            pos[prefix_sum] = 1;
+            prefix_sum += t.size() + 1;
+        }
+
+        text_begin    = sdsl::sd_vector(pos);
+        text_begin_ss = sdsl::select_support_sd<1>(&text_begin);
+        text_begin_rs = sdsl::rank_support_sd<1>(&text_begin);
+
+        sdsl::int_vector<8> tmp_text(concatenated_size - 1); // last text in collection needs no delimiter
+
+        uint8_t delimiter = alphabet_size_v<char_type> == 256 ? 255 : alphabet_size_v<char_type> + 1;
+
+        std::vector<uint8_t> tmp = text
+                                   | view::filter([&] (auto && t) { return !std::empty(t); })
+                                   | view::deep{view::to_rank}
+                                   | view::deep{view::transform([] (uint8_t const r) { return r + 1; })}
+                                   | ranges::view::join(delimiter);
```

Comment:
ah, I see that you have different functions for construct so you would need to apply my lambda from above here, too (and can replace the `is_collection` inside the lambda).